tensorflow/compiler/xla/service/depthwise_convolution_converter.cc

```diff
@@ -0,0 +1,212 @@
+
+  // When mapping depthwise conv backward filter to batch grouped convolution, 
+  // tf2xla bridge needs to swap the output batch and feature dimension. Since we 
+  // want to use grouped convolution APIs, this swap needs to be reverted.
+  dim_numbers.set_output_batch_dimension(output_feature_dimension);
+  dim_numbers.set_output_feature_dimension(output_batch_dimension);
+
+  if (!is_cost_viable_(convolution)) {
+    Shape transformed_filter_grad_shape = SwapInputOutputFeatureDims(
+        convolution->shape(), dim_numbers.output_batch_dimension(),
+        dim_numbers.output_feature_dimension());
+
+    int64 num_groups = convolution->batch_group_count();
+    int64 input_batch_dimension = dim_numbers.input_batch_dimension();
+    int64 input_batch = lhs->shape().dimensions(input_batch_dimension);
+    int64 input_feature_dimension = dim_numbers.input_feature_dimension();
+    int64 input_feature = lhs->shape().dimensions(input_feature_dimension);
+
+    CHECK_EQ(input_batch, num_groups)
+        << "Feature group count should be equal to number of input features "
+           "for depthwise convolution";
+
+    // Reshape batch_dim C -> [G, C/G] - Batch and feature dims have been
+    // swapped in tf2xla bridge
+    std::vector<int64> reshape_dims = lhs->shape().dimensions();
+    reshape_dims[input_batch_dimension] =
+        reshape_dims[input_batch_dimension] / num_groups;
+    reshape_dims.insert(reshape_dims.begin() + input_batch_dimension,
+                        num_groups);
+    lhs = add(HloInstruction::CreateReshape(
+        ShapeUtil::MakeShape(lhs->shape().element_type(), reshape_dims), lhs));
+
+    // Transpose G to the axis before N, For eg: [G, C/G, H, W, N ] -> [C/G, H,
+    // W, G, N]
+    std::vector<int64> transpose_dims(lhs->shape().dimensions_size());
+    std::iota(transpose_dims.begin(), transpose_dims.end(), 0);
+    transpose_dims.erase(transpose_dims.begin() + input_batch_dimension);
+    transpose_dims.insert(transpose_dims.begin() + input_feature_dimension,
+                          input_batch_dimension);
+    std::vector<int64> transpose_reshape_dims = lhs->shape().dimensions();
+    transpose_reshape_dims.erase(transpose_reshape_dims.begin() +
+                                 input_batch_dimension);
+    transpose_reshape_dims.insert(
+        transpose_reshape_dims.begin() + input_feature_dimension, num_groups);
+    lhs = add(HloInstruction::CreateTranspose(
+        ShapeUtil::MakeShape(lhs->shape().element_type(),
+                             transpose_reshape_dims),
+        lhs, transpose_dims));
+
+    // Merge [G,N] -> [N*G]
+    Shape new_shape = lhs->shape();
+    new_shape.DeleteDimension(input_feature_dimension);
+    new_shape.set_dimensions(input_feature_dimension,
+                             input_feature * num_groups);
+    lhs = add(HloInstruction::CreateReshape(new_shape, lhs));
+
+    auto new_convolution = add(HloInstruction::CreateConvolve(
```

Comment:
It would be safer to use convolution->CloneWithNewOperands() and then set dim_numbers, feature_group_count and batch_group_count. Because otherwise some information about the convolution instruction can get lost.
This will require new setter methods for feature_group_count and batch_group_count in tensorflow/compiler/xla/service/hlo_instructions.h